### PR TITLE
Preproduction database restore

### PR DESCRIPTION
**Pull request by neillturner**
### Context

support restore and restore to point in time in  preproduction for DR testing

### Changes proposed in this pull request

add preprod to database_restore and database_ptr workflow

#### Restore Point in Time

gh workflow run "Restore database from point in time to new database server" -r database-restore -f environment=preproduction -f restore-time=2024-10-29T06:00:00

see https://github.com/DFE-Digital/apply-for-qualified-teacher-status/actions/runs/11666600131
#### Restore from backup 

gh workflow run "Restore database from Azure storage" -r database-restore -f environment=preproduction -f backup-file=afqts_pp_adhoc_2024-11-01.sql.gz

see https://github.com/DFE-Digital/apply-for-qualified-teacher-status/actions/runs/11666736478

### Link to Trello card
[AFQTS: database restore and ptr to preprod](https://trello.com/c/VddQ4ylq/2121-afqts-database-restore-and-ptr-to-preprod)

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by neillturner** [comment]
[AFQTS: database restore and ptr to preprod](https://trello.com/c/VddQ4ylq/2121-afqts-database-restore-and-ptr-to-preprod)

**Review by shaheislamdfe — APPROVED** [review]

File: .github/workflows/database-restore.yml [hunk 1]
```diff
@@ -45,7 +46,7 @@ jobs:
       - name: Set environment variables
         run: |
           source global_config/${{ inputs.environment }}.sh
-          tf_vars_file=${{ env.TF_VARS_PATH }}/${{ inputs.environment }}/variables.tfvars.json
+          tf_vars_file=${{ env.TF_VARS_PATH }}/${{ env.ENVIRONMENT }}/variables.tfvars.json
```
Comment (saliceti) [inline .github/workflows/database-restore.yml:49]: ${{ env.ENVIRONMENT }} is not set at this point
But $ENVIRONMENT should work

File: .github/workflows/database-restore.yml [hunk 2]
```diff
@@ -64,7 +65,7 @@ jobs:
         with:
           storage-account: ${{ env.STORAGE_ACCOUNT_NAME }}
           resource-group: ${{ env.RESOURCE_GROUP_NAME }}
-          app-name: ${{ env.SERVICE_NAME }}-${{ inputs.environment }}-web
+          app-name: ${{ env.SERVICE_NAME }}-${{ env.ENVIRONMENT }}-web
```
Comment (saliceti) [inline .github/workflows/database-restore.yml:68]: ENVIRONMENT is not set in GITHUB_ENV

File: .github/workflows/database-ptr.yml [hunk 3]
```diff
@@ -45,7 +46,7 @@ jobs:
       - name: Set environment variables
         run: |
           source global_config/${{ inputs.environment }}.sh
-          tf_vars_file=${{ env.TF_VARS_PATH }}/${{ inputs.environment }}/variables.tfvars.json
+          tf_vars_file=${{ env.TF_VARS_PATH }}/${{ env.ENVIRONMENT }}/variables.tfvars.json
```
Comment (saliceti) [inline .github/workflows/database-ptr.yml:49]: ${{ env.ENVIRONMENT }} is not set at this point
But $ENVIRONMENT should work

**Pull request closed without merging**